File: IOS/appSegmentedControl/appSegmentedControl/ViewController.m
//
//  ViewController.m
//  appSegmentedControl
//
//  Created by Jorge on 24/04/24.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.imagenesVehiculos.image = [UIImage imageNamed:@"carro1"];
}


- (IBAction)segmentChangedValue:(UISegmentedControl *)sender {
    
    switch (self.segmentAspecto.selectedSegmentIndex) {
        case 0:
            self.imagenesVehiculos.contentMode = UIViewContentModeScaleToFill;
            break;
        case 1:
            self.imagenesVehiculos.contentMode = UIViewContentModeScaleAspectFill;
            break;
        case 2:
            self.imagenesVehiculos.contentMode = UIViewContentModeScaleAspectFit;
            break;
            
        default:
            break;
    }
    
}

- (IBAction)segmentValueChanged:(UISegmentedControl *)sender {
    
    switch (self.segmentControlVehiculos.selectedSegmentIndex) {
        case 0:
            self.imagenesVehiculos.image = [UIImage imageNamed:@"BICI_2"];
            break;
        case 1:
            self.imagenesVehiculos.image = [UIImage imageNamed:@"moto1"];
            break;
        case 2:
            self.imagenesVehiculos.image = [UIImage imageNamed:@"carro1"];
            break;
        case 3:
            self.imagenesVehiculos.image = [UIImage imageNamed:@"camioneta2"];
            break;
        default:
            break;
    }
    
}
@end

File: IOS/SWIFT_IOS/appCalculadora/appCalculadora/ViewController.swift
//
//  ViewController.swift
//  appCalculadora
//
//  Created by Jorge Cisneros on 20/05/24.
//

import UIKit

class ViewController: UIViewController {
    @IBOutlet weak var labelNumero: UILabel!
    @IBOutlet weak var slider1: UISlider!
    @IBOutlet weak var stepper1: UIStepper!
    @IBOutlet weak var segmentoTipo: UISegmentedControl!
    @IBOutlet weak var textoSalida: UITextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        labelNumero.text = String(Int(slider1.value))
        
    }

    @IBAction func segmentoSeleccionado(_ sender: UISegmentedControl) {
        
        let indice : Int = sender.selectedSegmentIndex
        
        var numero : Int = Int(labelNumero.text!)!
        
        if indice == 0 {
            
            // Verificar la conversion de numeros binarios negativos
            if numero < 0 {
            
                numero *= -1
                numero = 256 - numero
                
            }
            
            let n = String(numero, radix: 2)
            
            textoSalida.text = n
            
        } else if indice == 1 {
            
            let n = String(numero, radix: 8)
            
            textoSalida.text = n
            
        } else if indice == 2 {
            
            let n = String(numero, radix: 16)
            
            textoSalida.text = n
            
        }
        
    }
    
    @IBAction func valorCambiado(_ sender: UISlider) {
        
        labelNumero.text = "\( String( describing: lround( Double( sender.value ) ) ) )"
        
        stepper1.value = Double(sender.value)
        
        self.segmentoSeleccionado(segmentoTipo)
        
    }
    
    @IBAction func stepperCambio(_ sender: UIStepper) {
        
        labelNumero.text = "\( String( describing: lround( Double( sender.value ) ) ) )"
        
        slider1.value = Float( Double( stepper1.value ) )
        
        self.segmentoSeleccionado( segmentoTipo )
        
    }
    
}

